src/Sylius/Behat/Context/Setup/PromotionContext.php

```diff
@@ -142,24 +138,20 @@ public function itGivesFixedDiscountToEveryOrder(PromotionInterface $promotion,
      */
     public function itGivesPercentageDiscountToEveryOrder(PromotionInterface $promotion, $discount)
     {
-        $action = $this->actionFactory->createPercentageDiscount($discount);
-        $promotion->addAction($action);
-
-        $this->objectManager->flush();
+        $this->createPercentagePromotion($promotion, $discount);
     }
 
     /**
      * @Given /^([^"]+) gives ("[^"]+") fixed discount to every order with quantity at least ([^"]+)$/
      */
-    public function itGivesFixedDiscountToEveryOrderWithQuantityAtLeast(PromotionInterface $promotion, $amount, $quantity)
-    {
-        $action = $this->actionFactory->createFixedDiscount($amount);
-        $promotion->addAction($action);
-
+    public function itGivesFixedDiscountToEveryOrderWithQuantityAtLeast(
+        PromotionInterface $promotion,
+        $discount,
+        $quantity
+    ) {
         $rule = $this->ruleFactory->createCartQuantity((int) $quantity);
```

Comment:
I see you've done such a transformer [here](https://github.com/Sylius/Sylius/pull/4604/files#diff-c407f869351ef048e7032a0735e88f8bR42), so to avoid conflicts and duplication, maybe I'll wait with this change until your PR merge (or add it later, in separate PR)?